src/Control/Carrier/Stack.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Control.Carrier.Stack (
+  -- * Stack carrier
+  StackC,
+  runStack,
+
+  -- * Re-exports
+  module X,
+) where
+
+import Control.Carrier.Reader
+import Control.Effect.Stack as X
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Trans (MonadTrans, lift)
+import Data.Text (Text)
+
+newtype StackC m a = StackC {runStackC :: ReaderC [Text] m a}
+  deriving (Functor, Applicative, Monad, MonadIO)
+
+runStack :: [Text] -> StackC m a -> m a
```

Comment:
[nit] we have lots of `runStack []`, prefer if we collapse the default value in signature.

Prefer:
```haskell
runStack :: StackC m a -> m a
runStack = runStack' []

runStack' :: [Text] -> StackC m a -> m a
runStack' initial = runReader initial . runStackC
```

Reply:
Addressed in https://github.com/fossas/fossa-cli/commit/fa2a7fd9b768bee9a068e55f77545e5683ba9107